classes/EventRegistry.php

```diff
@@ -165,24 +174,7 @@ protected function processFormMLFields($fields, $model)
 
             $type = array_get($config, 'type', 'text');
 
-            if ($type == 'text') {
-                $fields[$name]['type'] = 'mltext';
-            }
-            elseif ($type == 'textarea') {
-                $fields[$name]['type'] = 'mltextarea';
-            }
-            elseif ($type == 'richeditor') {
-                $fields[$name]['type'] = 'mlricheditor';
-            }
-            elseif ($type == 'markdown') {
-                $fields[$name]['type'] = 'mlmarkdowneditor';
-            }
-            elseif ($type == 'repeater') {
-                $fields[$name]['type'] = 'mlrepeater';
-            }
-            elseif ($type == 'mediafinder') {
-                $fields[$name]['type'] = 'mlmediafinder';
-            }
+            $fields[$name]['type'] = $typesMap[$type];
```

Comment:
You'd have to wrap this in a conditional to check to see if there is a mapped type for the current type before replacing it.